components/dashtable.js

```diff
@@ -32,12 +32,40 @@ export default function DashTable(props) {
 
   let studentData = Object.entries(props.data).map(([i]) => {
     let studentName = Object.keys(props.data[i])[0];
+
+    const thresholdTime = 604800000; // time of one week in milliseconds
```

Comment:
this could also be refactored into the same component

Reply:
We've modified this part and moved the activity-related code into the react component!